Makefile

```diff
@@ -10,13 +10,18 @@ ifeq (,$(wildcard Makefile.global))
 endif
 
 include Makefile.global
-
 all: extension
 
+
+#build columnar only
+columnar:
+	$(MAKE) -C src/backend/columnar all
 # build extension
-extension: $(citus_top_builddir)/src/include/citus_version.h
+extension: $(citus_top_builddir)/src/include/citus_version.h columnar
```

Comment:
I finally got to tinkering with this PR, first smoketest of a `make install` from the root failed on the linker:
```
Undefined symbols for architecture x86_64:
  "_CheckCitusVersion", referenced from:
      _ColumnarProcessUtility in columnar_tableam.o
      _alter_columnar_table_set in columnar_tableam.o
      _alter_columnar_table_reset in columnar_tableam.o
      _ColumnarTableDropHook in columnar_tableam.o
      _columnar_beginscan in columnar_tableam.o
      _columnar_index_fetch_begin in columnar_tableam.o
      _columnar_tuple_satisfies_snapshot in columnar_tableam.o
      ...
ld: symbol(s) not found for architecture x86_64
Undefined symbols for architecture x86_64:
  "_CheckCitusVersion", referenced from:
      _ColumnarProcessUtility in columnar_tableam.o
      _alter_columnar_table_set in columnar_tableam.o
      _alter_columnar_table_reset in columnar_tableam.o
      _ColumnarTableDropHook in columnar_tableam.o
      _columnar_beginscan in columnar_tableam.o
      _columnar_index_fetch_begin in columnar_tableam.o
      _columnar_tuple_satisfies_snapshot in columnar_tableam.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [columnar.so] Error 1
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [columnar] Error 2
make: *** Waiting for unfinished jobs....
make[1]: *** [columnar.so] Error 1
make: *** [install-columnar] Error 2
```

Is this a known issue? Or am I the first/only to run into this.

I am building on a mac, don't know if @jeff-davis has a similar setup,

When running `make extension` a similar linker error pops up:

```
Undefined symbols for architecture x86_64:
  "_CheckCitusVersion", referenced from:
      _ColumnarProcessUtility in columnar_tableam.o
      _alter_columnar_table_set in columnar_tableam.o
      _alter_columnar_table_reset in columnar_tableam.o
      _ColumnarTableDropHook in columnar_tableam.o
      _columnar_beginscan in columnar_tableam.o
      _columnar_index_fetch_begin in columnar_tableam.o
      _columnar_tuple_satisfies_snapshot in columnar_tableam.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [columnar.so] Error 1
make: *** [columnar] Error 2
```

Let me know if I should debug this further (eg. if I am the only one having this issue)

Reply:
Will be fixed when #5688 is merged.

Reply:
I believe this is resolved.